app.js

```diff
@@ -245,67 +297,74 @@ exports.shambaDivaMiddleware = async (event, context) => {
-                            request = await sendRequest(dataset, band, scale, start_date, end_date, geometry_array)
+                                console.log(request)
 
-                            console.log(request)
+                                if (request != undefined && request.Success != undefined) {
+                                    sleep(60000).then(async _ => {
+                                        geostats_data = await getGeostatsData()
 
-                            if (request != undefined && request.Success != undefined) {
-                                sleep(60000).then(async _ => {
-                                    geostats_data = await getGeostatsData()
+                                        if (parseInt(geostats_data) != 0) {
 
-                                    console.log(geostats_data)
+                                            console.log(geostats_data)
 
-                                    finalResult = await setFinalReferenceValue(parseInt(poolData.id), ethers.BigNumber.from(geostats_data))
+                                            finalResult = await setFinalReferenceValue(parseInt(poolData.id), ethers.BigNumber.from(geostats_data))
 
-                                    console.log(finalResult)
+                                            console.log(finalResult)
 
-                                })
+                                        }
+
+                                    })
+                                }
                             }
-                        }
-                    })
-                }
-                else {
+                        })
+                    }
+                    else {
 
-                    console.log('balance >= 1')
-                    request = await sendRequest(dataset, band, scale, start_date, end_date, geometry_array)
+                        console.log('balance >= 1')
+                        request = await sendRequest(agg_x, dataset, band, scale, start_date, end_date, geometry_array)
 
-                    console.log(request)
+                        console.log(request)
 
-                    if (request != undefined && request.Success != undefined) {
-                        sleep(60000).then(async _ => {
-                            geostats_data = await getGeostatsData()
+                        if (request != undefined && request.Success != undefined) {
+                            sleep(60000).then(async _ => {
+                                geostats_data = await getGeostatsData()
 
-                            console.log(geostats_data)
+                                console.log(geostats_data)
 
-                            finalResult = await setFinalReferenceValue(parseInt(poolData.id), ethers.BigNumber.from(geostats_data))
+                                finalResult = await setFinalReferenceValue(parseInt(poolData.id), ethers.BigNumber.from(geostats_data))
```

Comment:
Same comment as above. Best is we make the value to report, i.e. `ethers.BigNumber.from(geostats_data).add(1)` a variable that we will use hear and in the equivalent statement above. Otherwise you run the risk of updating it in one place and not in the other